driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -61,4 +63,29 @@
      */
     T reduce(T initialValue, BinaryOperator<T> in);
 
+    IntegerExpression size();
+
+    T elementAt(IntegerExpression i);
+
+    default T elementAt(final int i) {
+        return this.elementAt(of(i));
+    }
+
+    T first();
+
+    T last();
+
+    BooleanExpression contains(T contains);
+
+    ArrayExpression<T> concat(ArrayExpression<T> array);
+
+    ArrayExpression<T> slice(IntegerExpression start, IntegerExpression length);
+
+    default ArrayExpression<T> slice(final int start, final int length) {
+        return this.slice(of(start), of(length));
+    }
+
+    ArrayExpression<T> setUnion(ArrayExpression<T> set);
+
+    ArrayExpression<T> distinct();
```

Comment:
Perhaps union should exclude `set` in its name instead, and the docs should cover this?

The name `set` seems confusing because of setField, and asSet implies another type. I do not think `distinct` implies that items will remain in the same order, like for example `removeDuplicates` might.

Reply:
If we look at SQL, what we call `setUnion` is called `union` there. However, what is called `distinct` in SQL preserves the order, unlike what we call `distinct` here. I.e., `distinct` may be seen as implying order.

If you don't see a way to simply and clearly express the lack of order in the names, it's fine, neither do I.

As for `setUnion` vs `union` - given that we don't express the lack of order in the name of `distinct`, I am also fine with `union`.

Reply:
I confirmed that this does not preserve order even when there is only one set in the union:
```
        assertExpression(
                Arrays.asList(1, 2, 3),
                ofIntegerArray(3, 2, 1, 3, 3, 3).distinct(),
```
passes. I agree that given what you say, distinct will imply they remain in the same order for users familiar with SQL distinct. And even the [streams distinct](https://javadoc.scijava.org/Java/java/util/stream/Stream.html#distinct--) specifies that if it is ordered, it will remain ordered. But the two other points (setField is weird, implies other type) still seem compelling. Maybe some variation of "unique" instead?

Reply:
I am fine with `unique`. If you leave `distinct`, I can also live with that.